### DISCUSSION TITLE
[FEEDBACK] Remove the PKCE Mandate on Non-Public Clients

### DISCUSSION THREAD

**Discussion by felix-tewhatuora** [tewhatuora/api-standards#70] [General]
### Summary

Affected page: https://apistandards.digital.health.nz/api-security/SecuringAPIswithOAuth2andOpenIDConnect

PKCE is mandated on all Authorisation code flows for all clients. I fail to see the security benefit of using PKCE for non-public clients (e.g. clients that are not mobile app/desktop application/SPA). Is there a reason it MUST be used for all Authorisation code flows?

It is strongly recommended to briefly outline the advantage of PKCE, why we need it, and when to use it (when the authz code cannot be safely stored, SPA/mobile apps/desktop apps, etc.) . 

Happy to be challenged.

Cheers,
Felix

### Link to standards item

https://apistandards.digital.health.nz/api-security/SecuringAPIswithOAuth2andOpenIDConnect

### Which area of the standards does this apply to?

- [ ] Part A - API Concepts
- [X] Part B - API Security
- [ ] Part C - API Design and Development
- [ ] Part D - FHIR
- [ ] Community guidelines

**Comment by mathusmith** [comment]
Agree this needs to be a SHOULD
